### PR TITLE
Added Cosmos Db nested Support for array type of attributes

### LINKED ISSUE #1624: [Bug]: using multiple filters on a nested object field returns an access forbidden message
### What happened?

The changes have been made according to the remarks made in #1538 this works well when the query looks like this:

```gql
{
    assets(
      filter: {
        and: [
          { type: { eq: "asset" } }
          { collection: { eq: $domain } }
          { data: { container: { neq: $domain } } }
         ]
      }
    ) {
      items {
        __typename
        ...AssetFields
      }
    }
  }
```

however when i use multiple filters on the nested object the access forbidden error is returned

```gql
{
    assets(
      filter: {
        and: [
          { type: { eq: "asset" } }
          { collection: { eq: $domain } }
          { data: { container: { neq: $domain } } }
          { data: { path: { notContains: ".json" } } }
        ]
      }
    ) {
      items {
        __typename
        ...AssetFields
      }
    }
  }
```

### Version

0.7.6 (part of the swa cli)

### What database are you using?

CosmosDB NoSQL

### What hosting model are you using?

Static Web Apps (SWA)

### Which API approach are you accessing DAB through?

GraphQL

### Relevant log output

```Text
dataApi] fail: Azure.DataApiBuilder.Service.Startup[0]
[dataApi]          at Azure.DataApiBuilder.Service.Models.GQLFilterParser.Parse(IMiddlewareContext ctx, IInputField filterArgumentSchema, List`1 fields, BaseQueryStructure queryStructure) in /_/src/Service/Models/GraphQLFilterParsers.cs:line 159
[dataApi]          at Azure.DataApiBuilder.Service.Models.GQLFilterParser.ParseAndOr(IMiddlewareContext ctx, IInputField argumentSchema, IInputField filterArgumentSchema, List`1 fields, BaseQueryStructure baseQuery, PredicateOperation op) in /_/src/Service/Models/GraphQLFilterParsers.cs:line 430
[dataApi]          at Azure.DataApiBuilder.Service.Models.GQLFilterParser.Parse(IMiddlewareContext ctx, IInputField filterArgumentSchema, List`1 fields, BaseQueryStructure queryStructure) in /_/src/Service/Models/GraphQLFilterParsers.cs:line 56
[dataApi]          at Azure.DataApiBuilder.Service.Resolvers.CosmosQueryStructure.Init(IDictionary`2 queryParams) in /_/src/Service/Resolvers/CosmosQueryStructure.cs:line 168
[dataApi]          at Azure.DataApiBuilder.Service.Resolvers.CosmosQueryEngine.ExecuteAsync(IMiddlewareContext context, IDictionary`2 parameters) in /_/src/Service/Resolvers/CosmosQueryEngine.cs:line 64
[dataApi]          at Azure.DataApiBuilder.Service.Services.ResolverMiddleware.InvokeAsync(IMiddlewareContext context) in /_/src/Service/Services/ResolverMiddleware.cs:line 146
[dataApi]          at HotChocolate.Utilities.MiddlewareCompiler`1.ExpressionHelper.AwaitTaskHelper(Task task)
[dataApi]          at HotChocolate.AspNetCore.Authorization.AuthorizeMiddleware.InvokeAsync(IDirectiveContext context)
[dataApi]          at HotChocolate.Utilities.MiddlewareCompiler`1.ExpressionHelper.AwaitTaskHelper(Task task)
[dataApi]          at HotChocolate.Execution.Processing.Tasks.ResolverTask.ExecuteResolverPipelineAsync(CancellationToken cancellationToken)
[dataApi]          at HotChocolate.Execution.Processing.Tasks.ResolverTask.TryExecuteAsync(CancellationToken cancellationToken)
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #1628: [Bug]: Using filter on multi-level nested objects causes runtime error (GraphQL, CosmosDB)
### What happened?

**Simplified version of the schema looks like:**

```
type EmployeeAggregate @model {
  id: ID
  Payload: Payload
}

type Payload {
  Id: ID
  Jobs: [Job]
}

type Job {
  Fte: Int
}
```

**Config looks like:**
```
"entities": {
    "EmployeeAggregate": {
      "source": "employeeAggregates_v2",
      "permissions": [
        {
          "role": "anonymous",
          "actions": [
            "*"
          ]
        }
      ]
    },
    "Payload": {
      "source": "employeeAggregates_v2",
      "permissions": [
        {
          "role": "anonymous",
          "actions": [
            "*"
          ]
        }
      ]
    },
    "Job": {
      "source": "employeeAggregates_v2",
      "graphql": {
        "type": {
          "singular": "Job",
          "plural": "Jobs"
        }
      }
	}
}
```

**Query looks like:**
```query EmployeeAggregates {
    employeeAggregates(first: 10, filter: {Payload: {Jobs: {Fte: {eq: 1}}}}) {
        items {
            Payload {
                Id
            }
        }
    }
}
```

**Error looks like:**
```
{
    "errors": [
        {
            "message": "GraphQL type doesn't match any entity name or singular type in the runtime config.",
            "extensions": {
                "code": "BadRequest"
            }
        }
    ]
}
```

**Note**
Filtering on first-level nested object (e.g. "Payload") works. However, filtering second-level nested objects (e.g. "Jobs") fails.

### Version

0.7.6

### What database are you using?

CosmosDB NoSQL

### What hosting model are you using?

Static Web Apps (SWA)

### Which API approach are you accessing DAB through?

GraphQL

### Relevant log output

```Text
{
    "errors": [
        {
            "message": "GraphQL type doesn't match any entity name or singular type in the runtime config.",
            "extensions": {
                "code": "BadRequest"
            }
        }
    ]
}
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by sourabh1007**
## Why make this change?
Closes #1628 
Closes #1624
If document contains Array type of attribute. Filtering using that Array is not possible right now. 

## What is this change?
As part of this PR, adding support to filter for array type also.

## How was this tested?

- [x] Integration Tests
- [x] Unit Tests

## Sample Request(s)

Refer test cases.

### CODE REVIEW

**Review by neeraj-sharma2592 — APPROVED** [review]
LGTM

File: src/Core/Models/GraphQLFilterParsers.cs [hunk 1]
```diff
@@ -232,6 +251,88 @@ public GQLFilterParser(RuntimeConfigProvider runtimeConfigProvider, IMetadataPro
...
+        string entityType,
+        List<PredicateOperand> predicates,
+        CosmosQueryStructure queryStructure,
+        ISqlMetadataProvider metadataProvider)
+    {
+        string entityName = metadataProvider.GetEntityName(entityType);
+
+        // Validate that the field referenced in the nested input filter can be accessed.
+        bool entityAccessPermitted = queryStructure.AuthorizationResolver.AreRoleAndOperationDefinedForEntity(
+            entityIdentifier: entityName,
```
Comment (tarazou9) [inline src/Core/Models/GraphQLFilterParsers.cs:283]: Here the permission checking is using the "EntityName", in the caller of ```HandleNestedFilterForCosmos```, it's getting the parameter value by using ```string nestedFieldTypeName = fieldDefinitionNode.Type.NamedType().Name.Value;```  Last time that I remember, Cosmos allows to have a different GraphQL type name than the entity name in the config. and it uses the `model` directive to map between the two. So if the name originally provided doesn't match any entity name, will this permission check always fail?

Comment (sourabh1007) [reply src/Core/Models/GraphQLFilterParsers.cs:283]: @tarazou9 if you are talking about aliasing..
here is the config https://github.com/Azure/data-api-builder/blob/9da650e6b4f5545349935247aa84e58fc58e88d2/src/Service.Tests/dab-config.CosmosDb_NoSql.json#L429

Here is the test cases covering that: https://github.com/Azure/data-api-builder/blob/9da650e6b4f5545349935247aa84e58fc58e88d2/src/Service.Tests/CosmosTests/QueryFilterTests.cs#L161

**Review by tarazou9 — APPROVED** [review]

**Pull request merged**